_posts/2023-11-15-Codesgiving-Part-1.md

```diff
@@ -0,0 +1,289 @@
+Using a troubleshooting approach like the above can help you break down big challenges into bite-sized chunks.
+Consider each step as a "module" or "section" which needs to be addressed sequentially.
+
+### Embrace a Learning Mindset
+
+Open-source contributions almost always entail learning of some kind.
+Many contributions happen solely in the form of code and text communications which are easily misinterpreted.
+__Assume positive intent__ and accept input from others while upholding your own ideas to share successful contributions together.
+Prepare yourself by intentionally opening your mind to input from others, even if you're sure you're absolutely "right".
+
+### Setting Software Scheduling Expectations
+
+
+<script src="https://cdn.jsdelivr.net/npm/vega@5.22.1"></script>
+<script src="https://cdn.jsdelivr.net/npm/vega-lite@5.6.0"></script>
+<script src="https://cdn.jsdelivr.net/npm/vega-embed@6.21.0"></script>
+<div id="vis"></div>
+
+<script>
+  var spec = {
+    "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
+    "description": "Suggested Open-source Time Allocations",
+    "width": 300,
+    "height": 300,
+    "data": {
+      "values": [
+        {"task": "Planning", "percentage": 33},
+        {"task": "Coding", "percentage": 16},
+        {"task": "Component and System Testing", "percentage": 25},
+        {"task": "Code Review, Revisions, and Post-Actions", "percentage": 25}
+      ]
+    },
+    "mark": {"type": "arc", "tooltip": true},
+    "encoding": {
+      "theta": {"field": "percentage", "type": "quantitative", "title": "Percentage"},
+      "color": {"field": "task", "type": "nominal", "title": "Task"}
+    }
+  };
+
+  const embed_opt = {"mode": "vega-lite"};
+  const el = document.getElementById('vis');
+  const view = vegaEmbed("#vis", spec, embed_opt);
+</script>
+_Suggested ratio of time spent by type of work for an open-source contribution._
+{:.center}
+
+The following modified rule of thumb from [_The Mythical Man Month_](https://www.oreilly.com/library/view/mythical-man-month-the/0201835959/) can assist with how you structure your time for an open-source contribution.
+Notice the emphasis on planning and testing and keep these in mind as you progress (the actual programming time can be small if adequate time has been spent on planning).
+Notably, the original time fractions are modified here with the final quarter of the time spent suggested as code review, revisions, and post-actions.
+Planning for the time expense of the added code review and related elements assists with keeping a learning mindset throughout the process (instead of feeling like the review is a "tack-on" or "optional / supplementary").
+A good motto to keep in mind throughout this process is [_Festina lente_](https://en.wikipedia.org/wiki/Festina_lente), or __"Make haste, slowly."__ (take care to move thoughtfully and as slow as necessary to do thing correctly the first time).
+
+1. 1/3 planning (~33%)
```

Comment:
Perhaps move this to right below the chart, since it's directly related? Good on you for putting it in text, as these data visualizations are seldom accessible.

Reply:
Great idea, thanks! I'll move this as you mentioned.